server/datastore/mysql/software.go

```diff
@@ -3775,6 +3813,15 @@ func (ds *Datastore) ListHostSoftware(ctx context.Context, host *fleet.Host, opt
 				hydrateHostSoftwareRecordFromDb(softwareTitleRecord, softwareTitle)
 			}
 
+			// This happens when there is a software installed on the host but it is also a vpp record, so we want
+			// to grab the vpp data from the installed vpp record and merge it onto the software record
+			if installedVppRecord, ok := hostVPPInstalledTitles[softwareTitleRecord.ID]; ok {
+				softwareTitleRecord.VPPAppAdamID = installedVppRecord.VPPAppAdamID
+				softwareTitleRecord.VPPAppVersion = installedVppRecord.VPPAppVersion
```

Comment:
This only impacts the accodiated `app_store_app` API response, right? So we'll still get the version shown in inventory (which is accurate for what's actually installed on the host, rather than what _would be_ installed when initiating a new install) for these apps?

Triple-checking because if we overwrite the inventory version then we'll have the wrong value here.

Reply:
That should be correct. As far as I understand the version shown on the front end looks at `HostSoftwareInstalledVersion`. I added tests that now verify this.